src/client-agent/start_agent.c

```diff
@@ -351,7 +363,11 @@ STATIC bool agent_handshake_to_server(int server_id, bool is_startup) {
         if (recv_b > 0) {
             /* Id of zero -- only one key allowed */
             if (ReadSecMSG(&keys, buffer, cleartext, 0, recv_b - 1, &msg_length, agt->server[server_id].rip, &tmp_msg) != KS_VALID) {
-                mwarn(MSG_ERROR, agt->server[server_id].rip);
+                if (strncmp(buffer, "#unrecognized", 13) == 0) {
```

Comment:
I think that `#unauthorized` would be more straightforward; it implies that the connection was rejected for authentication or permission reasons. This is very common in security contexts.

Reply:
Changed!